unity-game/Assets/Scripts/ElevatorScripts/AutoElevator.cs

```diff
@@ -0,0 +1,220 @@
+﻿using UnityEngine;
+using System.Collections;
+
+public class AutoElevator : MonoBehaviour
+{
+    public bool notMoving;
+    public bool reachedDestination;
+    public bool onStartPoint;
+    public bool onEndPoint;
+
+    public GameObject objectToMove;
+    public GameObject theRightWall;
+    public GameObject theLeftWall;
+    public GameObject theStartLeftBarrier;
+    public GameObject theStartRightBarrier;
+    public GameObject theEndLeftBarrier;
+    public GameObject theEndRightBarrier;
+    public GameObject theStartPoint;
+    public GameObject theEndPoint;
+
+    public Transform startPoint;
+    public Transform endPoint;
+
+    public float moveSpeed;
+    public float waitTime;
+    public float waitCounter;
+    
+    private Vector3 currentTarget;
+
+
+    void Start()
+    {
+        currentTarget = endPoint.position;
+
+        waitCounter = waitTime;
+
+        theLeftWall.SetActive(false);
+        theRightWall.SetActive(false);
+        theStartLeftBarrier.SetActive(false);
+        theStartRightBarrier.SetActive(false);
+        theEndLeftBarrier.SetActive(false);
+        theEndRightBarrier.SetActive(false);
+    }
+
+
+
+    void Update()
+    {     
+        if (objectToMove.transform.position == endPoint.position)
+        {
+            onStartPoint = false;
+            onEndPoint = true;
+            if (waitCounter < 0f)
+            {
+                reachedDestination = true;
+                theStartLeftBarrier.SetActive(false);
+                theStartRightBarrier.SetActive(false);
+                theEndLeftBarrier.SetActive(false);
+                theEndRightBarrier.SetActive(false);
+            }
+            notMoving = true;
+            currentTarget = startPoint.position;
+            if (theEndPoint.gameObject.GetComponent<ElevatorBookends>().leftWall == true)
```

Comment:
You're referring to "if (theEndPoint.gameObject.GetComponent<ElevatorBookends>().leftWall == true)"?